config.lua

```diff
@@ -25,24 +29,66 @@ config({'development', 'production'}, {
     -- read & write permissions to that path.
     store_path = 'store',
 
-     -- for sending email
-    mail_server = "",
-    mail_user     = "",
-    mail_password = "",
-    mail_from = "",
-    mail_footer = ""
+    -- for sending email
+    mailgun = {
+        domain = os.getenv('MAILGUN_DOMAIN'),
+        api_key = os.getenv('MAILGUN_API_KEY')
+    },
+    mail_user = os.getenv('MAIL_SMTP_USER'),
+    mail_password = os.getenv('MAIL_SMTP_PASSWORD'),
+    mail_server = os.getenv('MAIL_SMTP_SERVER'),
+    mail_from_name = 'Snap!Cloud',
+    mail_from = "noreply@snap-cloud.cs10.org",
+    mail_footer = "This is a test",
+
+    measure_performance = true,
+
+    cors_domains = {
+        'localhost:' .. port,
+        'snap.berkeley.edu'
+    }
 })
 
 config('production', {
-    site_name = 'Snap Cloud',
+    use_daemon = 'on',
     postgres = {
         host = os.getenv('DATABASE_URL'),
         user = os.getenv('DATABASE_USERNAME'),
         password = os.getenv('DATABASE_PASSWORD'),
         database = os.getenv('DATABASE_NAME')
     },
+    site_name = 'Snap Cloud',
+    hostname = 'snap-clous.cs10.org',
+    ssl_cert_name = os.getenv('SSL_CERT_NAME') or 'snap-cloud.cs10.org',
+    enable_https = true,
     secret = os.getenv('SESSION_SECRET_BASE'),
     num_workers = 12,
     code_cache = 'on',
-    store_path = '/opt/snap'
+    store_path = '/opt/snap',
+
+    --- TODO: See if we can turn this on without a big hit
+    measure_performance = false,
+
+    cors_domains = {
+        'snap.berkeley.edu',
+        -- MIRRORS
+        'cs10.org',
+        'bjc.edc.org',
+        'web.media.mit.edu',
+        'byob.eecs.berkeley.edu',
+        -- RESEARCH PROJECTS
+        'eliza.csc.ncsu.edu',
+        'lambda.cs10.org',
+        -- EDX (sad)
+        'courses.edge.edx.org',
+        'courses.edx.org',
+        'd37djvu3ytnwxt.cloudfront.net',
+        'preview.courses.edge.edx.org',
+        'preview.courses.edx.org',
+        'preview.edge.edx.org',
+        'preview.edx.org',
+        'studio.edge.edx.org',
+        'studio.edx.org',
+        'edge.edx.org'
+    }
```

Comment:
Each query needs to check for domains, so it's crucial to make the test as fast as possible, that's why it's better to have the domains be the keys to a table where all contents are booleans.

Reply:
I didn't see you had done this. But I built the string once only on app load.

So there's just a string of multiple domains that gets returned with each request, instead of checking each time. 

I'm not sure if either approach is better or worse. (Mine sends more data, but is less checking, but yours is the opposite.)

Reply:
I've gone and just used yours for now. 

I think, ultimately, stuff like the CORS list is a best left as an app configuration, but it's minor. :)